test/e2e/framework/service/jig.go

```diff
@@ -82,6 +98,33 @@ func NewTestJig(client clientset.Interface, namespace, name string) *TestJig {
 	return j
 }
 
+// GetUnusedStaticNodePortAndReserve reserves first free port in static range and returns it
```

Comment:
we also need to modify https://github.com/kubernetes/kubernetes/blob/a0ad420018d69601136223f91b818a73747a7ce3/test/e2e/framework/service/jig.go#L478-L497
 to use this new method

Reply:
Thanks for pointing this out. Yes, looks like flaky conditions can occur here as well. Since 
`should be able to change the type and ports of a UDP service` and `should be able to change the type and ports of a TCP service` are the tests that use this function, should I go ahead and address it in this PR ?

Reply:
I believe current behaviour of this function would be trying to allocate a nodeport from _dynamic_ range until a free nodeport is found.

Reply:
yeah, my question is that if is used in some tests that can lead to the same problem you are solving here, should not we try to use the static range to avoid it?